File: data/code_pairs/fortran/865.f90
module foo_mod
  implicit none
  private
  public :: foo, foo2

  type :: foo
    integer :: i
  contains
    procedure, pass :: doit
    procedure, pass :: getit
  end type foo

  type, extends(foo) :: foo2
    integer :: j
  contains
    procedure, pass :: doit => doit2
    procedure, pass :: getit => getit2
  end type foo2

contains
  subroutine doit(this)
    class(foo), intent(inout) :: this
    this%i = 1
  end subroutine doit

  function getit(this) result(res)
    class(foo), intent(in) :: this
    integer :: res
    res = this%i
  end function getit

  subroutine doit2(this)
    class(foo2), intent(inout) :: this
    this%i = 2
    this%j = 3
  end subroutine doit2

  function getit2(this) result(res)
    class(foo2), intent(in) :: this
    integer :: res
    res = this%j
  end function getit2

end module foo_mod

File: data/code_pairs/cpp/865.cpp
#ifndef FOO_H
#define FOO_H

// Base class
class Foo {
public:
    virtual void doit() { i = 1; }
    virtual int getit() const { return i; }
protected:
    int i{0};
};

// Derived class
class Foo2 : public Foo {
public:
    void doit() override {
        i = 2;
        j = 3;
    }
    int getit() const override { return j; }
private:
    int j{0};
};

#endif // FOO_H